### DISCUSSION TITLE
Unable to install in Unity 2019.4.13f1

### DISCUSSION THREAD

**Discussion by DSotirCW** [JoshClose/CsvHelper#1725] [Q&A]
Ive been trying to get CSVHelper installed in Unity today and am having some major issues. I have tried in Rider with NuGet packages at first but that doesnt seem to work. I found online I can just unzip the NuGet package and drag the .dll into Unitys Plugins folder but that gives me the error below. I have Unitys API level set to .NET Standard 2.0 and am using the netstandard 2.0 .dll. Any help would be appreciated!

`Assembly 'Assets/Plugins/CsvHelper.dll' will not be loaded due to errors:
Unable to resolve reference 'Microsoft.Bcl.AsyncInterfaces'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'System.Threading.Tasks.Extensions'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
Unable to resolve reference 'Microsoft.Bcl.HashCode'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
`

**Comment by DSotirCW** [comment]
@JoshClose If you could help me out I would greatly appreciate it.

**Reply by JoshClose** [reply]
How are you installing it?

**Reply by DSotirCW** [reply]
I am downloading the package from your repo, unzipping it, and grabbing the netstandard 2.0 .dll file and putting that into Unity. It seems like using version 12.3.2 works, but anything later (13+) doesnt work and shows that error i posted.

**Reply by JoshClose** [reply]
There are dependencies that you would need to download too. Click the "Dependencies" here. https://www.nuget.org/packages/CsvHelper/25.0.0

I would suggest using NuGet for Unity so it'll resolve all the dependencies for you. https://github.com/GlitchEnzo/NuGetForUnity

**Reply by DSotirCW** [reply]
Why is it that the 12.3.2 seems to work?

**Reply by JoshClose** [reply]
The dependency didn't exist in that version.

**Reply by DSotirCW** [reply]
Ahh okay. If I dont need anything super up to date with it, is there anything wrong with staying on that version? Im just using it for some simple CSV parsing really

**Reply by JoshClose** [reply]
No the old version is fine. I would still suggest using NuGetForUnity though. ;)

**Reply by Ro-tico** [reply]
This really helped me out. Thank you!

**Reply by Ro-tico** [reply]
Hi JoshClose, I experience a lot of problems with Nuget once I have installed CV-Helper. A buddy of mine was also dealing with this and he managed to do it much easier by making a directory in assets called Plugins and put all the dependencies of CV-Helper in this directory. Download the Zip file for all the dependencies. The only dependency that needs to be deleted is the one called Miscrosoft.C-Sharp as this is already implemented in Unity. This is much more user-friendly. 

Cheers
[CsvHelper Plugins.zip](https://github.com/JoshClose/CsvHelper/files/7101944/CsvHelper.Plugins.zip)

**Reply by JoshClose** [reply]
Thanks for the info.
